// @line/bot-sdk/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhook = exports.shop = exports.moduleAttach = exports.moduleOperation = exports.messagingApi = exports.manageAudience = exports.liff = exports.insight = exports.channelAccessToken = exports.OAuth = exports.validateSignature = exports.middleware = exports.Client = void 0;
const client_1 = require("./client");
exports.Client = client_1.default;
Object.defineProperty(exports, "OAuth", { enumerable: true, get: function () { return client_1.OAuth; } });
const middleware_1 = require("./middleware");
exports.middleware = middleware_1.default;
const validate_signature_1 = require("./validate-signature");
exports.validateSignature = validate_signature_1.default;
// re-export exceptions and types
__exportStar(require("./exceptions"), exports);
__exportStar(require("./types"), exports);
const channelAccessToken = require("./channel-access-token/api");
exports.channelAccessToken = channelAccessToken;
const insight = require("./insight/api");
exports.insight = insight;
const liff = require("./liff/api");
exports.liff = liff;
const manageAudience = require("./manage-audience/api");
exports.manageAudience = manageAudience;
const messagingApi = require("./messaging-api/api");
exports.messagingApi = messagingApi;
// Note: `module` is reserved word in Javascript.
const moduleOperation = require("./module/api");
exports.moduleOperation = moduleOperation;
const moduleAttach = require("./module-attach/api");
exports.moduleAttach = moduleAttach;
const shop = require("./shop/api");
exports.shop = shop;
const webhook = require("./webhook/api");
exports.webhook = webhook;
