// @line/bot-sdk/dist/manage-audience/api/manageAudienceClient.js
"use strict";
/**
 * LINE Messaging API
 * This document describes LINE Messaging API.
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManageAudienceClient = void 0;
const Types = require("../../types");
const utils_1 = require("../../utils");
const http_1 = require("../../http");
class ManageAudienceClient {
    constructor(config) {
        if (!config.baseURL) {
            config.baseURL = "https://api.line.me";
        }
        this.httpClient = new http_1.default({
            defaultHeaders: {
                Authorization: "Bearer " + config.channelAccessToken,
            },
            responseParser: this.parseHTTPResponse.bind(this),
            baseURL: config.baseURL,
        });
    }
    parseHTTPResponse(response) {
        const { LINE_REQUEST_ID_HTTP_HEADER_NAME } = Types;
        let resBody = Object.assign({}, response.data);
        if (response.headers[LINE_REQUEST_ID_HTTP_HEADER_NAME]) {
            resBody[LINE_REQUEST_ID_HTTP_HEADER_NAME] =
                response.headers[LINE_REQUEST_ID_HTTP_HEADER_NAME];
        }
        return resBody;
    }
    /**
     * Activate audience
     * @param audienceGroupId The audience ID.
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#activate-audience-group"> Documentation</a>
     */
    async activateAudienceGroup(audienceGroupId) {
        const res = this.httpClient.put("/v2/bot/audienceGroup/{audienceGroupId}/activate".replace("{audienceGroupId}", String(audienceGroupId)));
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Add user IDs or Identifiers for Advertisers (IFAs) to an audience for uploading user IDs (by JSON)
     * @param addAudienceToAudienceGroupRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#update-upload-audience-group"> Documentation</a>
     */
    async addAudienceToAudienceGroup(addAudienceToAudienceGroupRequest) {
        const params = addAudienceToAudienceGroupRequest;
        const res = this.httpClient.put("/v2/bot/audienceGroup/upload", params);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Create audience for uploading user IDs (by JSON)
     * @param createAudienceGroupRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#create-upload-audience-group"> Documentation</a>
     */
    async createAudienceGroup(createAudienceGroupRequest) {
        const params = createAudienceGroupRequest;
        const res = this.httpClient.post("/v2/bot/audienceGroup/upload", params);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Create audience for click-based retargeting
     * @param createClickBasedAudienceGroupRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#create-click-audience-group"> Documentation</a>
     */
    async createClickBasedAudienceGroup(createClickBasedAudienceGroupRequest) {
        const params = createClickBasedAudienceGroupRequest;
        const res = this.httpClient.post("/v2/bot/audienceGroup/click", params);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Create audience for impression-based retargeting
     * @param createImpBasedAudienceGroupRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#create-imp-audience-group"> Documentation</a>
     */
    async createImpBasedAudienceGroup(createImpBasedAudienceGroupRequest) {
        const params = createImpBasedAudienceGroupRequest;
        const res = this.httpClient.post("/v2/bot/audienceGroup/imp", params);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Delete audience
     * @param audienceGroupId The audience ID.
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#delete-audience-group"> Documentation</a>
     */
    async deleteAudienceGroup(audienceGroupId) {
        const res = this.httpClient.delete("/v2/bot/audienceGroup/{audienceGroupId}".replace("{audienceGroupId}", String(audienceGroupId)));
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Gets audience data.
     * @param audienceGroupId The audience ID.
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#get-audience-group"> Documentation</a>
     */
    async getAudienceData(audienceGroupId) {
        const res = this.httpClient.get("/v2/bot/audienceGroup/{audienceGroupId}".replace("{audienceGroupId}", String(audienceGroupId)));
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Get the authority level of the audience
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#get-authority-level"> Documentation</a>
     */
    async getAudienceGroupAuthorityLevel() {
        const res = this.httpClient.get("/v2/bot/audienceGroup/authorityLevel");
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Gets data for more than one audience.
     * @param page The page to return when getting (paginated) results. Must be 1 or higher.
     * @param description The name of the audience(s) to return. You can search for partial matches. This is case-insensitive, meaning AUDIENCE and audience are considered identical. If omitted, the name of the audience(s) will not be used as a search criterion.
     * @param status The status of the audience(s) to return. If omitted, the status of the audience(s) will not be used as a search criterion.
     * @param size The number of audiences per page. Default: 20 Max: 40
     * @param includesExternalPublicGroups true (default): Get public audiences created in all channels linked to the same bot. false: Get audiences created in the same channel.
     * @param createRoute How the audience was created. If omitted, all audiences are included.  `OA_MANAGER`: Return only audiences created with LINE Official Account Manager (opens new window). `MESSAGING_API`: Return only audiences created with Messaging API.
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#get-audience-groups"> Documentation</a>
     */
    async getAudienceGroups(page, description, status, size, includesExternalPublicGroups, createRoute) {
        const queryParams = {
            page: page,
            description: description,
            status: status,
            size: size,
            includesExternalPublicGroups: includesExternalPublicGroups,
            createRoute: createRoute,
        };
        const res = this.httpClient.get("/v2/bot/audienceGroup/list", queryParams);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Change the authority level of the audience
     * @param updateAudienceGroupAuthorityLevelRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#change-authority-level"> Documentation</a>
     */
    async updateAudienceGroupAuthorityLevel(updateAudienceGroupAuthorityLevelRequest) {
        const params = updateAudienceGroupAuthorityLevelRequest;
        const res = this.httpClient.put("/v2/bot/audienceGroup/authorityLevel", params);
        return (0, utils_1.ensureJSON)(res);
    }
    /**
     * Renames an existing audience.
     * @param audienceGroupId The audience ID.
     * @param updateAudienceGroupDescriptionRequest
     *
     * @see <a href="https://developers.line.biz/en/reference/messaging-api/#set-description-audience-group"> Documentation</a>
     */
    async updateAudienceGroupDescription(audienceGroupId, updateAudienceGroupDescriptionRequest) {
        const params = updateAudienceGroupDescriptionRequest;
        const res = this.httpClient.put("/v2/bot/audienceGroup/{audienceGroupId}/updateDescription".replace("{audienceGroupId}", String(audienceGroupId)), params);
        return (0, utils_1.ensureJSON)(res);
    }
}
exports.ManageAudienceClient = ManageAudienceClient;
